src/components/navigation/ListItemLink.tsx

```diff
@@ -7,56 +7,52 @@ import {
     useMediaQuery,
     useTheme,
 } from '@mui/material';
-import PropTypes from 'prop-types';
-import { forwardRef } from 'react';
-import { NavLink } from 'react-router-dom';
+import { forwardRef, ReactNode } from 'react';
+import { NavLink, NavLinkProps } from 'react-router-dom';
 import { Widths } from '../../AppLayout';
 
-const ListItemLinkProps = {
-    badgeContent: PropTypes.number,
-    disabled: PropTypes.bool,
-    icon: PropTypes.element,
-    isOpen: PropTypes.bool,
-    link: PropTypes.string.isRequired,
-    menuWidth: PropTypes.number,
-    title: PropTypes.string.isRequired,
-};
+interface ListItemLinkProps {
```

Comment:
Personal preference would be to group required and optional properties, with required properties taking precedence.